telosnetwork/telos-wallet
# warn user and remove telos cloud wallet support

# Fixes #685

### blocked by: https://github.com/telosnetwork/telos-wallet/issues/687
Somehow the local storage gets cleared and we lose not only the flag for the notification "don't show me this again" but also all contracts and NFTs metadata cached from the indexer.

## Description
Because the Telos Cloud service will be discontinued, we must warn users with a notification and at the same time program the login button to auto-remove after the specific date (Dic 31th)

## Test scenarios
### Test the notification
- go to [this link](https://deploy-preview-686--wallet-develop-mainnet.netlify.app/)
  - you should see a new notification type saying what's going on with Telos Cloud and a checkbox
- Dismiss without touching the checkbox
- Refresh the page
  - the notification should appear again
- Click on checkbox and then dissmiss
- reload the page
  - the notification should not appear anymore

### Test the scheduled action
- run this branch locally
```bash
git fetch --all; git checkout 685-remove-telos-cloud-wallet-support-on-dec-15th; yarn; yarn dev
```
- change the year from 2023 to 2022 in the file: `src/App.vue`
```typescript
export const isTodayBeforeTelosCloudDown = new Date().getTime() < new Date('2023-12-31').getTime();
```
- go to http://localhost:8080/
  - you should not see the notification
- try to connect your wallet
  - you should not see the Telos Cloud button

## Checklist:
